packages/toolkit/src/query/createApi.ts

```diff
@@ -236,15 +236,18 @@ export function buildCreateApi<Modules extends [Module<any>, ...Module<any>[]]>(
       })
     )
 
+    const { serializeQueryArgs, registerArgsSerializerForEndpoint } =
+      buildSerializeQueryArgs(options.serializeQueryArgs)
+
     const optionsWithDefaults = {
       reducerPath: 'api',
-      serializeQueryArgs: defaultSerializeQueryArgs,
       keepUnusedDataFor: 60,
       refetchOnMountOrArgChange: false,
       refetchOnFocus: false,
       refetchOnReconnect: false,
       ...options,
       extractRehydrationInfo,
+      serializeQueryArgs,
```

Comment:
If you inline a `serializeQueryArgs` function here instead, it can access `context.endpointDefinitions[endpointName].serializeQueryArgs` here and you won't need to implement a `registerArgsSerializerForEndpoint` function and maintain that extra data.